dc-cudami-editor/src/components/modals/ImageAdderModal.jsx

```diff
@@ -0,0 +1,161 @@
+import {FontAwesomeIcon} from '@fortawesome/react-fontawesome'
+import {faCode, faUpload} from '@fortawesome/free-solid-svg-icons'
+import {publish, subscribe} from 'pubsub-js'
+import React, {Component} from 'react'
+import {
+  ListGroup,
+  ListGroupItem,
+  Modal,
+  ModalBody,
+  ModalHeader,
+} from 'reactstrap'
+import {withTranslation} from 'react-i18next'
+
+import ImageMetadataForm from './imageAdder/ImageMetadataForm'
+import ImageRenderingHintsForm from './imageAdder/ImageRenderingHintsForm'
+import ImageUploadForm from './imageAdder/ImageUploadForm'
+import ImageUrlForm from './imageAdder/ImageUrlForm'
+
+class ImageAdderModal extends Component {
+  constructor(props) {
+    super(props)
+    this.initialAttributes = {
+      alignment: 'left',
+      altText: '',
+      caption: '',
+      label: '',
+      linkNewTab: true,
+      linkUrl: '',
+      title: '',
+      url: '',
+      width: '33%',
+    }
+    this.state = {
+      attributes: this.initialAttributes,
+      readOnly: false,
+      step: 'initial',
+    }
+    subscribe('editor.show-image-modal', () => {
+      this.props.onToggle()
+    })
+  }
+
+  addImageToEditor = () => {
+    this.toggle()
+    const filteredAttributes = Object.entries(this.state.attributes).reduce(
```

Comment:
Mit etwas weniger Zeremonie:
```suggestion
    const filteredAttributes = Object.fromEntries(
      Object.entries(this.state.attributes)
            .filter(([key, value] => value))
```

Reply:
Super, schaut viel übersichtlicher aus!